### PR TITLE
minor YouTube URL code refactoring

### PR DESCRIPTION
**Pull request by 2is10**
@ahsu1230

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/common/util.js [hunk 1]
```diff
@@ -70,8 +66,9 @@ angular.module('util', [])
            (/Trident/.exec(navigator.userAgent) != null))
         );
       },
-      isYoutubeUrl: function (url) {
-        return getYoutubeIdFromUrl(url);
+      getYoutubeIdFromUrl: function (url) {
+        var match = url.match(youtubeVideoUrlRe);
+        return match && match[1];
       },
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/common/util.js:72]: The name `isYoutubeUrl` sets the expectation that the function will return `true`/`false`. I think it’s important for function names to convey or hint at their return types. We could go with `getYoutubeIdFromUrlIfTheUrlIsIndeedAYouTubeVideoUrlOrElseJustReturnNull` but I’ve compromised.

File: kifi-backend/modules/shoebox/angular/src/keep/keep.js [hunk 2]
```diff
@@ -120,16 +120,9 @@ angular.module('kifi')
           return text;
         }
 
-        function isYoutubeCard(url) {
-          var youtubeId = util.isYoutubeUrl(url);
-          if (youtubeId) {
-            scope.youtubeId = youtubeId;
-          }
-          return scope.youtubeId;
-        }
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js:129]: One rule of thumb that I’ve really come to value over the years is that a function should _either_ return a value _or_ have a side effect—not both. It makes for much clearer code.

Notice that `isYoutubeCard` was both returning a value and modifying `scope.youtubeId`. The side effect was surprising given the predicate-style name.

Comment (joshm1) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js:129]: :+1:

File: kifi-backend/modules/shoebox/angular/src/common/util.js (hunk 1 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/common/util.js:72]: lol =)

File: kifi-backend/modules/shoebox/angular/src/keep/keep.js (hunk 2 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js:129]: :+1:  I see

**Comment by ahsu1230** [comment]
Thanks for the cleanup! looks a lot better =)

File: kifi-backend/modules/shoebox/angular/src/keep/keep.js (hunk 2 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js:129]: Very nice.

**Pull request merged**